prefq/server.py

```diff
@@ -114,25 +124,39 @@ def receive_feedback():
 
     print("\n\nServer: Starting receive_feedback() [...]")
 
-    print("Server: Receiving feedback")
+    # should be replaced by flask session object
+    # pylint: disable=global-statement
+    # pylint: disable=global-variable-not-assigned
+    global feedback_data
+    global n_pending_queries
+
     data = flask.request.json  # Represents incoming client http request in json format
 
     # Extract received JSON data
     is_left_preferred = data["is_left_preferred"]
-    video_filename_left = data["video_filename_left"]
-    video_index_left = int(video_filename_left.split(".")[0]) - 1
-    batch_index = video_index_left // 2
+    left_filename = data["video_filename_left"]
+
+    query_id = left_filename[: -len("-left.webm")]
```

Comment:
This is imitation specific. Instead, the client should explicitly add a query ID to the request. The imitation client can extract the ID from the filename, but the server does not need to be aware of this.

Reply:
this function refers to the FeedbackClient, not the ImitationClient. Therefore, changes would need to be adjusted in the .html/.js files, & within the server function "serve_video()", Information about the Query ID and the suffix name convention would have to be passed by modifying the default flask.send_from_directory(folder,filename) function, that the server uses, to send videos to the FeedbackClient.

how about instead adding a flask session variables:

app.config["LEFT_FILENAME_SUFFIX"]="-left.webm"
app.config["RIGHT_FILENAME_SUFFIX"]="-right.webm"

and (dis)assemble filenames accordingly? 

this way we can add flexibility without changing the defaults

Reply:
As discussed in the meeting, we can send the query ID in the request (omit the filenames) and then have the server save the files using the "-left.webm", "-right.webm" convention. I think we don't need to make this configurable for now. The imitation client can then extract the ID from the filename, the static client (`query_client.py`) can generate an ID from the compared indices (e.g., `"01-04"`).

Reply:
The Query Client now sends Query IDs instead of filenames, then the Server assambles the filenames locally. However, either within the Feedback Client or within the Server Script, the Query ID needs to be extracted from the filename in order to avoid unnessessary overhead.

Reply:
Okay, I think relying on the name convention internally is fine since we are in control of the filenames. Thanks for changing this!